### ISSUE TITLE
[reproducibility] add util function to enhance reproducibility

### ISSUE THREAD

**Issue by eric-haibin-lin** [dmlc/gluon-nlp#1110]
There are a few seeds that may lead to nondeterministic training loss: 
```
random.seed(seed)
mx.random.seed(seed)
numpy.random.seed(seed)
```
Sometimes our users miss one of them. Shall we add an API `gluonnlp.utils.seed` to set the seed for all of them?

**Comment by sxjscience** [comment]
We should add one. Actually, even if all these are set, we cannot guarantee reproducibility due to the issues here: https://github.com/apache/incubator-mxnet/issues/16705.

However, we still should add `gluonnlp.utils.seed` and enhance the MXNet side to ensure reproducibility (or teach the user how to tweak the environmental variables for reproducibility concerns).

**Comment by avinashsai** [comment]
Hi @eric-haibin-lin 
I can take this up. Can you provide me little more context on how the function should be??

**Comment by avinashsai** [comment]
closed via #1151

**Comment by szha** [comment]
@avinashsai thanks for the contribution
